test/sparse/sparse.jl

```diff
@@ -1,469 +1,518 @@
+    @test sparse(Any[1,2,3], Any[1,2,3], Any[1,1,1], 5, 4) == sparse([1,2,3], [1,2,3], [1,1,1], 5, 4)
+end
+
 se33 = speye(3)
 do33 = ones(3)
-@test isequal(se33 * se33, se33)
-
-# check sparse binary op
-@test all(Array(se33 + convert(SparseMatrixCSC{Float32,Int32}, se33)) == 2*eye(3))
-@test all(Array(se33 * convert(SparseMatrixCSC{Float32,Int32}, se33)) == eye(3))
-
-# check horiz concatenation
-@test all([se33 se33] == sparse([1, 2, 3, 1, 2, 3], [1, 2, 3, 4, 5, 6], ones(6)))
-
-# check vert concatenation
-@test all([se33; se33] == sparse([1, 4, 2, 5, 3, 6], [1, 1, 2, 2, 3, 3], ones(6)))
-se33_32bit = convert(SparseMatrixCSC{Float32,Int32}, se33)
-@test all([se33; se33_32bit] == sparse([1, 4, 2, 5, 3, 6], [1, 1, 2, 2, 3, 3], ones(6)))
-
-# check h+v concatenation
-se44 = speye(4)
-sz42 = spzeros(4, 2)
-sz41 = spzeros(4, 1)
-sz34 = spzeros(3, 4)
-se77 = speye(7)
-@test all([se44 sz42 sz41; sz34 se33] == se77)
-
-# check blkdiag concatenation
-@test all(blkdiag(se33, se33) == sparse([1, 2, 3, 4, 5, 6], [1, 2, 3, 4, 5, 6], ones(6)))
-
-# check concatenation promotion
-sz41_f32 = spzeros(Float32, 4, 1)
-se33_i32 = speye(Int32, 3, 3)
-@test all([se44 sz42 sz41_f32; sz34 se33_i32] == se77)
-
-# check mixed sparse-dense concatenation
-sz33 = spzeros(3, 3)
-de33 = eye(3)
-@test  all([se33 de33; sz33 se33] == Array([se33 se33; sz33 se33 ]))
-
-# check splicing + concatenation on
-# random instances, with nested vcat
-# also side-checks sparse ref
-for i = 1 : 10
-    a = sprand(5, 4, 0.5)
-    @test all([a[1:2,1:2] a[1:2,3:4]; a[3:5,1] [a[3:4,2:4]; a[5:5,2:4]]] == a)
-end
-
-# sparse ref
+
+@testset "Matrix operations" begin
+    @test isequal(se33 * se33, se33)
+end
+
+@testset "Sparse binary operation" begin
+    se33 = speye(3)
+    do33 = ones(3)
+
+    @test all(Array(se33 + convert(SparseMatrixCSC{Float32,Int32}, se33)) == 2*eye(3))
+    @test all(Array(se33 * convert(SparseMatrixCSC{Float32,Int32}, se33)) == eye(3))
+end
+
+@testset "Concatenation tests" begin
+    @testset "Horizontal concatenation" begin
+        @test all([se33 se33] == sparse([1, 2, 3, 1, 2, 3], [1, 2, 3, 4, 5, 6], ones(6)))
+    end
+
+    @testset "Vertical concatenation" begin
+        @test all([se33; se33] == sparse([1, 4, 2, 5, 3, 6], [1, 1, 2, 2, 3, 3], ones(6)))
+        se33_32bit = convert(SparseMatrixCSC{Float32,Int32}, se33)
+        @test all([se33; se33_32bit] == sparse([1, 4, 2, 5, 3, 6], [1, 1, 2, 2, 3, 3], ones(6)))
+    end
+
+    se44 = speye(4)
+    sz42 = spzeros(4, 2)
+    sz41 = spzeros(4, 1)
+    sz34 = spzeros(3, 4)
+    se77 = speye(7)
+    @testset "h+v concatenation" begin
+        @test all([se44 sz42 sz41; sz34 se33] == se77)
+    end
+
+    @testset "Blkdiag concatenation" begin
+        @test all(blkdiag(se33, se33) == sparse([1, 2, 3, 4, 5, 6], [1, 2, 3, 4, 5, 6], ones(6)))
+    end
+
+    @testset "Concatenation promotion" begin
+        sz41_f32 = spzeros(Float32, 4, 1)
+        se33_i32 = speye(Int32, 3, 3)
```

Comment:
Preserve the "(non-square)", or perhaps "rectangular"? Nix the now-redundant comment?